test/unit/hl7_validator_test.rb

```diff
@@ -9,52 +9,43 @@
   end
 
   describe 'Validating a good resource' do
-    it "Shouldn't pass back any messages" do
-      patient = FHIR::Patient.new
-      stub_request(:post, "#{@validator_url}/validate")
-        .with(
-          query: { profile: 'http://hl7.org/fhir/StructureDefinition/Patient' },
-          body: patient.source_contents
-        )
-        .to_return(status: 200, body: load_fixture('validator_good_response'))
-      result = @validator.validate(patient, FHIR)
-
-      assert_empty result[:errors]
-      assert_empty result[:warnings]
-      assert_empty result[:information]
+    before do
+      @resource = FHIR::CapabilityStatement.new
+      @profile = FHIR::Definitions.resource_definition(@resource.resourceType).url
     end
 
-    it 'Should reject code-invalid issues' do
+    it "Shouldn't pass back any messages" do
       patient = FHIR::Patient.new
       stub_request(:post, "#{@validator_url}/validate")
         .with(
           query: { profile: 'http://hl7.org/fhir/StructureDefinition/Patient' },
           body: patient.source_contents
         )
-        .to_return(status: 200, body: load_fixture('validator_invalid_code_response'))
+        .to_return(status: 200, body: load_fixture('validator_good_response'))
       result = @validator.validate(patient, FHIR)
 
       assert_empty result[:errors]
       assert_empty result[:warnings]
       assert_empty result[:information]
     end
-  end
 
-  describe 'Validating a bad resource' do
-    it 'Should pass back an error message' do
-      patient = FHIR::Patient.new(gender: 'robot')
+    it 'removes excluded errors' do
+      outcome = load_fixture('hl7_validator_operation_outcome')
 
-      stub_request(:post, "#{@validator_url}/validate")
+      stub_request(:post, @validator_url + '/validate')
         .with(
-          query: { profile: 'http://hl7.org/fhir/StructureDefinition/Patient' },
-          body: patient.source_contents
+          query: { 'profile': @profile },
+          body: @resource.source_contents
+        )
+        .to_return(
+          status: 200,
+          body: outcome
         )
-        .to_return(status: 200, body: load_fixture('validator_bad_response'))
-      result = @validator.validate(patient, FHIR)
 
-      assert_equal 2, result[:errors].size
-      assert_equal 1, result[:warnings].size
-      assert_equal 1, result[:information].size
+      result = @validator.validate(@resource, FHIR, @profile)
+      assert result[:errors].length == 2
```

Comment:
The benefit of `assert_equal` here over `assert` is that the error message is more descriptive (so it can say `expected 1 but was given 3` instead of `expected false but was given true`).